mattermost-plugin/server/boards/boardsapp_util.go

```diff
@@ -45,6 +47,11 @@ func createBoardsConfig(mmconfig mm_model.Config, baseURL string, serverID strin
 	if mmconfig.FileSettings.AmazonS3Trace != nil {
 		filesS3Config.Trace = *mmconfig.FileSettings.AmazonS3Trace
 	}
+	if mmconfig.FileSettings.AmazonS3RequestTimeoutMilliseconds == nil {
+		filesS3Config.Timeout = defaultS3Timeout
+	} else {
+		filesS3Config.Timeout = *mmconfig.FileSettings.AmazonS3RequestTimeoutMilliseconds
+	}
```

Comment:
We can protect as well from a zero config value, in case somebody thinks that that would disable the timeout completely:

```suggestion
	filesS3Config.Timeout = defaultS3Timeout
	if mmconfig.FileSettings.AmazonS3RequestTimeoutMilliseconds != nil && *mmconfig.FileSettings.AmazonS3RequestTimeoutMilliseconds != 0 {
		filesS3Config.Timeout = *mmconfig.FileSettings.AmazonS3RequestTimeoutMilliseconds
	}
```

Reply:
Such sense, much wow.

Reply:
On it.

Reply:
Done! Protected from 0 and negative timeout